src/utils/wallet.js

```diff
@@ -438,53 +647,266 @@ class Wallet {
             ...options,
-            ...(await this.signatureFor(this.accountId))
+            ...(await this.signatureFor(tempTwoFactorAccount ? tempTwoFactorAccount : this))
         });
     }
 
     async initializeRecoveryMethod(accountId, method) {
-        return await this.postSignedJson('/account/initializeRecoveryMethod', {
+        const tempAccount = getTempAccount()
+        if (tempAccount && tempAccount.accountId) {
+            return await sendJson('POST', ACCOUNT_HELPER_URL + '/account/initializeRecoveryMethodForTempAccount', {
+                accountId,
+                method,
+            });
+        } else {
+            return await this.postSignedJson('/account/initializeRecoveryMethod', {
+                accountId,
+                method
+            });
+        }
+    }
+
+    async validateSecurityCode(accountId, method, securityCode) {
+        const tempAccount = getTempAccount()
+        if (tempAccount && tempAccount.accountId) {
+            return await sendJson('POST', ACCOUNT_HELPER_URL + '/account/validateSecurityCodeForTempAccount', {
+                accountId,
+                method,
+                securityCode
+            });
+        } else {
+            return await this.postSignedJson('/account/validateSecurityCode', {
+                accountId,
+                method,
+                securityCode
+            });
+        }
+    }
+
+    async initTwoFactor(accountId, method) {
+        // clear any previous requests in localStorage (for verifyTwoFactor)
+        setRequest({})
+        return await this.postSignedJson('/2fa/init', {
             accountId,
             method
         });
     }
 
-    async validateSecurityCode(accountId, method, securityCode) {
-        return await this.postSignedJson('/account/validateSecurityCode', {
+    async reInitTwoFactor(accountId, method) {
+        // clear any previous requests in localStorage (for verifyTwoFactor)
+        setRequest({})
+        return this.sendTwoFactor(accountId, method)
+    }
+
+    async sendTwoFactor(accountId, method, requestId = -1, data = {}) {
+        if (!accountId) accountId = this.accountId
+        if (!method) method = await this.get2faMethod()
+        // add request to local storage
+        setRequest({ accountId, requestId, data })
+
+        console.log(accountId, requestId, data)
+        console.log('tempTwoFactorAccount', tempTwoFactorAccount)
+
+        const serverResponse = this.postSignedJson('/2fa/send', {
             accountId,
             method,
-            securityCode
+            requestId,
+            data
+        }).catch((e) => console.log('/2fa/send failed', e));
+        
+        if (requestId !== -1) {
+            // we know the requestId of what we want to confirm so pop the modal now and wait on that
+            return new Promise((resolve) => {
+                store.dispatch(promptTwoFactor((verified) => {
```

Comment:
let's avoid utils/wallet depending on store (should be the other way around). If smth needs to access store it probably needs to be either of Redux action, reducer or React component.

Reply:
handled in twoFactor.js now and updated